### PR TITLE
Check getSlot bounds during interact capture, fixes #602

### LINKED ISSUE #602: Click Interact Inventory Event triggers IndexOutOfBoundsException
```
java.util.concurrent.ExecutionException: java.lang.IndexOutOfBoundsException: Index: 49, Size: 45
        at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_71]
        at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_71]
        at net.minecraft.util.Util.func_181617_a(SourceFile:45) [g.class:?]
        at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:660) [MinecraftServer.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:344) [ko.class:?]
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:605) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:481) [MinecraftServer.class:?]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_71]
Caused by: java.lang.IndexOutOfBoundsException: Index: 49, Size: 45
        at java.util.ArrayList.rangeCheck(ArrayList.java:653) ~[?:1.8.0_71]
        at java.util.ArrayList.get(ArrayList.java:429) ~[?:1.8.0_71]
        at net.minecraft.inventory.Container.func_75139_a(Container.java:111) ~[xi.class:?]
        at org.spongepowered.common.event.SpongeCommonEventFactory.handleClickInteractInventoryEvent(SpongeCommonEventFactory.java:306) ~[SpongeCommonEventFactory.class:1.8.9-1763-4.1.0-BETA-0]
        at org.spongepowered.common.event.CauseTracker.handlePostTickCaptures(CauseTracker.java:496) ~[CauseTracker.class:1.8.9-1763-4.1.0-BETA-0]
        at org.spongepowered.common.network.PacketUtil.onProcessPacket(PacketUtil.java:108) ~[PacketUtil.class:1.8.9-1763-4.1.0-BETA-0]
        at net.minecraft.network.PacketThreadUtil$1.redirect$onProcessPacket$0(SourceFile:51) ~[fh$1.class:?]
        at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) ~[fh$1.class:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_71]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_71]
        at net.minecraft.util.Util.func_181617_a(SourceFile:44) ~[g.class:?]
        ... 5 more
```

### PR DESCRIPTION
**Pull request by modwizcode**
This is a proper fix at the source of the problem, this overrides https://github.com/SpongePowered/SpongeCommon/pull/610.

### CODE REVIEW

File: src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java [hunk 1]
```diff
@@ -302,7 +303,8 @@ public static void handleClickInteractInventoryEvent(Cause cause, EntityPlayerMP
         Transaction<ItemStackSnapshot> cursorTransaction = new Transaction<>(StaticMixinHelper.lastCursor, newCursor);
         ClickInventoryEvent clickEvent = null;
         // Handle empty slot clicks
-        if (((IMixinContainer) player.openContainer).getCapturedTransactions().size() == 0 && packetIn.getSlotId() >= 0) {
+        if (((IMixinContainer) player.openContainer).getCapturedTransactions().size() == 0 && packetIn.getSlotId() >= 0
+                && packetIn.getSlotId() < player.openContainer.inventorySlots.size()) {
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java]: Will this allow for negative slot ids? If so, it shouldn't.

File: src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java [hunk 2]
```diff
@@ -259,7 +259,8 @@ public static void handleCreativeClickInventoryEvent(Cause cause, EntityPlayerMP
                     entitySnapshotBuilder.build(), (org.spongepowered.api.item.inventory.Container) player.openContainer, (World) player.worldObj,
                     ((IMixinContainer) player.openContainer).getCapturedTransactions());
         } else {
-            if (((IMixinContainer) player.openContainer).getCapturedTransactions().size() == 0 && packetIn.getSlotId() >= 0) {
+            if (((IMixinContainer) player.openContainer).getCapturedTransactions().size() == 0 && packetIn.getSlotId() >= 0
+                    && packetIn.getSlotId() < player.openContainer.inventorySlots.size()) {
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java]: Will this allow for negative slot ids? If so, it shouldn't.

File: src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java (hunk 1 above)
Comment (modwizcode) [inline src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java]: No? It already checks `packIn.getSlotId() >= 0`?

File: src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java (hunk 2 above)
Comment (modwizcode) [inline src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java]: Same as above

File: src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java (hunk 1 above)
Comment (kashike) [inline src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java]: Sorry, I was totally blind there - I thought the diff removed that check.

**Pull request merged**